<a href="https://colab.research.google.com/github/emgrimm42/NLP-Hausarbeit/blob/main/Analyse_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Ziel dieses Dokuments
---
Ziel dieses Dokuments ist ein Notebook zu erstellen in dem ich die ersten Analyse Schritte durchführen kann. Diese sind zunächst einmal die benutzung des Wortes Lügenpresse rauszusortieren, bzw. die Artikel zu sortieren, in Artikel die das Wort Lügenpresse (auch andere Schreibweisen) enthalten und in Artikel die das Wort Lügenpresse nicht enthalten.
Da ich unteranderem die zeitliche Nutzung von Lügenpresse einordnen möchte glaube ich nicht dass es zum jetzigen Zeitpunkt mega notwendig ist die Tabellenform in der meine Daten sich bisher befinden aufzulösen.
Angesichts der wiederholten Probleme die ich auf Google Colab hatte weil meine Daten zu groß waren um sie darzustellen würde es vielleicht Sinn machen eine zweite Excel Datei anzulegen in der nur Artikel mit dem Wort Lügenpresse zu finden sind.

1. Schritt: Liste mit Lügenpresse Schreibweisen (zur durchsuchung)
2. Schritt: Erstellen einer neuen kleineren Datei, oder potenziell dem erneuten Runterladen der Daten aus dem Zeitungsportal, falls das einfacher ist.
3. Schritt: Liste mit Erscheinungsdaten aus der kleineren Datei
4. Schritt: Visualisieren dieser Daten - Zeiträume vordefinieren oder nicht?





# Analyse Plan:
---
2 Teilige Analyse
1. Teil: Benutzung des Wortes Lügenpresse
Nur Lügenpresse da es ja um die Etablierung des Wortes geht.

Suche nach dem Wort bzw. anderen Schreibweisen innerhalb des existierenden Corpuses.

Einsortierung der Nutzungen, auf welche Personen/Kontexte beziehen sich diese Nennungen (brauch eine klar definierte Liste an Personen/Kontexten

2. Teil: Nutzung von Synonymen
Lügenpresse war ja ein vergleichsweiser junge Begriff, d.h. es ist möglich dass gleichbedeutende Synonyme noch weiter verbreitet waren.

Liste an Synonymen in verschiedene (Meinungs) Gruppen einteilen

Durch Suche nach einzelnen Gruppen ermitteln welche Gruppen in dieser konkreten Zeitung am besten vertreten waren.

## Lügenpresse Schreibweisen
- Lügenpresse
- Luegenpresse
- Lugenpresse
- Lügenpress
- Lügenprese
- Luegenpress
- Luegenprese
- Lugenpress
- Lugenprese
- Lügenpreffe (s / f Verwechslung)
- Lugenpreffe
- Lügen Presse
- Lügenpreſſe
-


# Filtern des Datensatzes
---
Der nächste Schritt ist den Datensatz auf die eben erstellten Lügenpresse Schreibweisen zu filtern.

In [ ]:
import pandas as pd
import string
import nltk
from nltk.corpus import stopwords
from collections import defaultdict, Counter
import matplotlib.pyplot as plt
import numpy as np
from wordcloud import WordCloud
import plotly.graph_objects as go

In [ ]:
# Import the data as Excel file. Make sure the path to your excel file is correct, including ''.
df = pd.read_excel('/content/Volltext_2751569-2.xlsx')

# Display the first few rows of the DataFrame to verify it's loaded correctly.
df.head(5)

,Unnamed: 0,id,pagenumber,paper_title,provider_ddb_id,provider,zdb_id,publication_date,place_of_distribution,language,thumbnail,pagefulltext,pagename,preview_reference,plainpagefulltext
0,0,IKB4OLLQYFCWSKJE6UJM2QLQ5CAO2RXB-FILE_0001_DDB...,1,Deutsche Zeitung,PE423JPDSCU6C72BAC2PUBOHAINDRGFO,Bayerische Staatsbibliothek,2751569-2,1848-03-08T12:00:00Z,"['Heidelberg', 'Leipzig', 'Frankfurt am Main',...",['ger'],34a5b6cf-d665-4a3f-bb70-2322c7fca675,/data/altos/IK/B4/IKB4OLLQYFCWSKJE6UJM2QLQ5CAO...,FILE_0001_DDB_FULLTEXT,https://api.deutsche-digitale-bibliothek.de/bi...,Alle Poſtämternehmen Be ellnn Die Haupterpedit...
1,1,IKB4OLLQYFCWSKJE6UJM2QLQ5CAO2RXB-FILE_0002_DDB...,2,Deutsche Zeitung,PE423JPDSCU6C72BAC2PUBOHAINDRGFO,Bayerische Staatsbibliothek,2751569-2,1848-03-08T12:00:00Z,"['Heidelberg', 'Leipzig', 'Frankfurt am Main',...",['ger'],34a5b6cf-d665-4a3f-bb70-2322c7fca675,/data/altos/IK/B4/IKB4OLLQYFCWSKJE6UJM2QLQ5CAO...,FILE_0002_DDB_FULLTEXT,https://api.deutsche-digitale-bibliothek.de/bi...,"5Z8 Mittelpunkt, den es nur zu ſtärken, aber n..."
2,2,IKB4OLLQYFCWSKJE6UJM2QLQ5CAO2RXB-FILE_0003_DDB...,3,Deutsche Zeitung,PE423JPDSCU6C72BAC2PUBOHAINDRGFO,Bayerische Staatsbibliothek,2751569-2,1848-03-08T12:00:00Z,"['Heidelberg', 'Leipzig', 'Frankfurt am Main',...",['ger'],34a5b6cf-d665-4a3f-bb70-2322c7fca675,/data/altos/IK/B4/IKB4OLLQYFCWSKJE6UJM2QLQ5CAO...,FILE_0003_DDB_FULLTEXT,https://api.deutsche-digitale-bibliothek.de/bi...,539 beſtimmter Sprache um: die Verleihung eine...
3,3,IKB4OLLQYFCWSKJE6UJM2QLQ5CAO2RXB-FILE_0004_DDB...,4,Deutsche Zeitung,PE423JPDSCU6C72BAC2PUBOHAINDRGFO,Bayerische Staatsbibliothek,2751569-2,1848-03-08T12:00:00Z,"['Heidelberg', 'Leipzig', 'Frankfurt am Main',...",['ger'],34a5b6cf-d665-4a3f-bb70-2322c7fca675,/data/altos/IK/B4/IKB4OLLQYFCWSKJE6UJM2QLQ5CAO...,FILE_0004_DDB_FULLTEXT,https://api.deutsche-digitale-bibliothek.de/bi...,540 Sp. Z.; 1) Unverzügliche Berufung der Stän...
4,4,IKB4OLLQYFCWSKJE6UJM2QLQ5CAO2RXB-FILE_0005_DDB...,5,Deutsche Zeitung,PE423JPDSCU6C72BAC2PUBOHAINDRGFO,Bayerische Staatsbibliothek,2751569-2,1848-03-08T12:00:00Z,"['Heidelberg', 'Leipzig', 'Frankfurt am Main',...",['ger'],34a5b6cf-d665-4a3f-bb70-2322c7fca675,/data/altos/IK/B4/IKB4OLLQYFCWSKJE6UJM2QLQ5CAO...,FILE_0005_DDB_FULLTEXT,https://api.deutsche-digitale-bibliothek.de/bi...,"541 in Bruchſal, in Neckarbiſchofsheim, auch i..."


In [ ]:
# Filter the DataFrame to include only rows where 'plainpagefulltext' contains 'Lügenpresse'.
df_Lügenpresse = df[df['plainpagefulltext'].str.contains('Lügenpresse', na=False)]

# Display the filtered DataFrame.
df_Lügenpresse.head()

,Unnamed: 0,id,pagenumber,paper_title,provider_ddb_id,provider,zdb_id,publication_date,place_of_distribution,language,thumbnail,pagefulltext,pagename,preview_reference,plainpagefulltext


Okay, sieht so aus als ob das wort Lügenpresse selbst so nicht in den Artikeln vorkommt, aber zur Sicherheit Checken wir noch mal mit einem Begriff der wie wir wissen definitiv im Datensatz vorkommt. Laut einer vorherigen Analyse kommt das Wort 'deut fchen' 1008 Mal in dem Datensatz vor.

In [ ]:
df_test = df[df['plainpagefulltext'].str.contains('deutſchen', na=False)]
print(len(df_test))

5132


Zwar ist 5132 deutlich über 1008, aber dass kann daher komen dass so Begriffe wie Deutschen Zeitung in den 1008 nich inkludiert waren. Dieser schnelle Test hat auf jeden Fall bewiesen dass der Code funktioniert. Nun muss ich ihn noch für alle vorhim erittelten Schreibweisen laufen lassen, am besten einmal mit s und einmal mit ſ.

In [ ]:
keywords = ['Lügenpresse', 'Luegenpresse', 'Lugenpresse', 'Lügenpress', 'Lügenprese', 'Luegenpress', 'Luegenprese', 'Lugenpress', 'Lugenprese', 'Lügen Presse',
 'Lügenpreſſe', 'Luegenpreſſe', 'Lugenpreſſe', 'Lügenpreſſ', 'Lügenpreſe', 'Luegenpreſſ', 'Luegenpreſe', 'Lugenpreſſ', 'Lugenpreſe', 'Lügen Preſſe',
 'lügenpreſſe', 'luegenpreſſe', 'lugenpreſſe', 'lügenpreſſ', 'lügenpreſe', 'luegenpreſſ', 'luegenpreſe', 'lugenpreſſ', 'lugenpreſe', 'lügen Preſſe']

pattern = '|'.join(keywords)

df_test = df[df['plainpagefulltext'].str.contains(pattern, na=False)]
print(len(df_test))

0


Okay, es sieht so aus als ob die Deutsche Zeitung den Begriff Lügenpresse in keiner mir erdenklichen Schreibweise genutzt hätte. Aber auch das könnte eine wichtige Erkentnis für eine Analyse sein.
Imerhin war der Begriff Lügenpresse an sich in den 1840ern noch nicht sonderlich verbreitet und stark religiös konservativ geprägt. Dementsprechen ist es nicht mega überraschend dass eine bürgerlich-liberale Zeitung wie die Deutsche Zeitung den Begriff selbst nicht benutzte.
Daher werde ich jetzt umschwingen und nach Synonymen für Lügenpresse suchen, mit denen die selbe Thematik vielleicht mit anderen Begriffen diskutiert wurde.

In [ ]:
synonyme = ["Lüge",
    "Lügenpresse",
    "Falschmeldung",
    "Lügenblatt",
    "Lügenblätter",
    "Hetzblatt",
    "Hetzpresse",
    "Schmierblatt",
    "Schmierpresse",
    "Pamphletpresse",
    "Parteipresse",
    "Demagogenpresse",
    "Revolutionspresse",
    "radikale Presse",
    "ultraradikale Presse",
    "jüdische Presse",
    "Judenpresse",
    "liberale Presse",
    "Freimaurerpresse",
    "Logenpresse",
    "Pressunfug",
    "Wühlerpresse",
    "Gegenpresse",
    "Zeitungsgeschrei",
    "Journalistenpack",
    "Tintenkleckserei"
]

pattern = '|'.join(synonyme)

df_test = df[df['plainpagefulltext'].str.contains(pattern, na=False)]
print(len(df_test))

361


Wir sehen also dass das Thema durchaus diskutiert wurde, nur halt dass dafür nicht der Begriff Lügenpresse benutzt wurde. Die drei letzten Begriffe sind satirische, also mal schauen wie viele der 361 Nennungen satirisch waren.

In [ ]:
satirisch = [ "Zeitungsgeschrei",
    "Journalistenpack",
    "Tintenkleckserei"]

pattern = '|'.join(satirisch)

df_test = df[df['plainpagefulltext'].str.contains(pattern, na=False)]
print(len(df_test))

0
